app/src/main/java/co/nimblehq/compose/crypto/ui/navigation/AppNavigation.kt

```diff
@@ -24,7 +30,23 @@ fun AppNavigation(
         composable(AppDestination.CoinDetail) {
             DetailScreen(
                 navigator = { destination -> navController.navigate(destination) },
-                coinId = it.arguments?.getString(KEY_COIN_ID).orEmpty()
+                coinId = it.arguments?.getString(KEY_COIN_ID).orEmpty(),
+                onNetworkReconnected = { callback ->
+                    onCallBackChange(callback)
+                }
+            )
+        }
+
+        dialog(AppDestination.NoNetwork.route) {
+            AppDialogPopUp(
+                onDismiss = { navController.popBackStack() },
+                onClick = {
+                    navController.popBackStack()
+                    globalDialogCallback()
```

Comment:
This would be better for readability, we will know it's the callback method rather than a local function.

```suggestion
                    globalDialogCallback.invoke()
```